test/unit/firebase-app.spec.ts

```diff
@@ -149,6 +155,70 @@ describe('FirebaseApp', () => {
       (mockApp.options as any).foo = 'changed';
       expect(mockApp.options).to.deep.equal(original);
     });
+
+    it('should overwrite the config values with the ones in the config file', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config.json';
+      const app = firebaseNamespace.initializeApp(mocks.appOptionsNoDatabaseUrl, mocks.appName);
+      expect(app.options.databaseURL).to.equal("https://hipster-chat.firebaseio.mock");
+      expect(app.options.projectId).to.equal("hipster-chat-mock");
+      expect(app.options.storageBucket).to.equal("hipster-chat.appspot.mock");
+    });
+
+    it('should throw when the environment variable points to non existing file', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_non_existant.json';
+      
+      expect(() => {
+       const app = firebaseNamespace.initializeApp(mocks.appOptionsNoDatabaseUrl, mocks.appName);
+      }).to.throw(`Failed to read app options file: Error: ENOENT: no such file or directory, open './test/resources/firebase_config_non_existant.json'`);
+    });
+
+    it('should throw when the environment variable points to non parsable file', () => {
+      process.env[FIREBASE_CONFIG_FILE_VAR] = './test/resources/firebase_config_bad.json'  ;
```

Comment:
Is this file empty? I would recommend adding some obviously malformed content in there to make that obvious.